contracts/src/Zappers/Modules/Exchanges/HybridCurveUniV3Exchange.sol

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.18;
+
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+import "openzeppelin-contracts/contracts/utils/math/Math.sol";
+
+import "../../../Interfaces/IWETH.sol";
+import "../../LeftoversSweep.sol";
+// Curve
+import "./Curve/ICurveStableswapNGPool.sol";
+// UniV3
+import "./UniswapV3/ISwapRouter.sol";
+import "./UniswapV3/IQuoterV2.sol";
+
+import "../../Interfaces/IExchange.sol";
+
+// import "forge-std/console2.sol";
+
+contract HybridCurveUniV3Exchange is LeftoversSweep, IExchange {
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable collToken;
+    IBoldToken public immutable boldToken;
+    IERC20 public immutable USDC;
+    IWETH public immutable WETH;
+
+    // Curve
+    ICurveStableswapNGPool public immutable curvePool;
+    uint128 public immutable USDC_INDEX;
+    uint128 public immutable BOLD_TOKEN_INDEX;
+
+    // Uniswap
+    uint24 public immutable feeUsdcWeth;
+    uint24 public immutable feeWethColl;
+    ISwapRouter public immutable uniV3Router;
+
+    constructor(
+        IERC20 _collToken,
+        IBoldToken _boldToken,
+        IERC20 _usdc,
+        IWETH _weth,
+        // Curve
+        ICurveStableswapNGPool _curvePool,
+        uint128 _usdcIndex,
+        uint128 _boldIndex,
+        // UniV3
+        uint24 _feeUsdcWeth,
+        uint24 _feeWethColl,
+        ISwapRouter _uniV3Router
+    ) {
+        collToken = _collToken;
+        boldToken = _boldToken;
+        USDC = _usdc;
+        WETH = _weth;
+
+        // Curve
+        curvePool = _curvePool;
+        USDC_INDEX = _usdcIndex;
+        BOLD_TOKEN_INDEX = _boldIndex;
+
+        // Uniswap
+        feeUsdcWeth = _feeUsdcWeth;
+        feeWethColl = _feeWethColl;
+        uniV3Router = _uniV3Router;
+    }
+
+    // Bold -> USDC on Curve; then USDC -> WETH, and optionally WETH -> Coll, on UniV3
+    function swapFromBold(uint256 _boldAmount, uint256 _minCollAmount) external returns (uint256) {
+        InitialBalances memory initialBalances;
+        _setHybridExchangeInitialBalances(initialBalances);
+
+        // Curve
+        ICurveStableswapNGPool curvePoolCached = curvePool;
+        boldToken.transferFrom(msg.sender, address(this), _boldAmount);
+        boldToken.approve(address(curvePoolCached), _boldAmount);
+
+        uint256 curveUsdcAmount = curvePoolCached.exchange(int128(BOLD_TOKEN_INDEX), int128(USDC_INDEX), _boldAmount, 0);
```

Comment:
I guess if this gets sandwiched the Curve pool may return much less USDC than desired, but ultimately the `_minCollAmount` ensures the caller gets what they want (or else it reverts)

Reply:
Yes, good point. I thought it would add unnecessarily complexity to try to fail earlier if we detect that the amount obtained in the first step is not going to be enough.